packages/libraries/core/src/components/button/ods-button-controller.ts

```diff
@@ -42,4 +42,37 @@ export class OdsButtonController extends OdsComponentController<OdsButton> {
       attribute: this.component.variant
     });
   }
+
+  handleClick(event: MouseEvent): void {
+    this.logger.log('Click on osds-button');
+
+    if (this.component.type) {
+      this.submitForm(this.component.type, event);
```

Comment:
you can simplify because of the `submitForm` method has already access to `type`:
```suggestion
      this.submitForm(event);
```